extensions/wikia/GlobalWatchlist/GlobalWatchlistBot.class.php

```diff
@@ -154,7 +154,6 @@ private function getUserObject( $userID ) {
 		if (\F::app()->wg->ExternalAuthType ) {
 			$mExtUser = ExternalUser::newFromId( $userID );
```

Comment:
Shouldn't we actually skip handling of `wgExternalAuthType` and rely on `User::newFromId` only?

I bet this `if` was needed for special cases like our internal wikis and Uncyclopedia separate user database.

Reply:
Per our conversation in the kitchen, the general idea behind `ExternalUser` is to encapsulate gluecode between MW's user schema and some external schema.  In my opinion, back in the past, it was a convenient and quick way for us to scale without digging to deep into MediaWiki.  Today the situation is different and "external" user database should not surprise anyone.  The transitional goal is to move all the logic to `User`, so that it handles data flow between MW and `wikicities.user`.  The ultimate goal is to refactor user to use an external microservice rather than database.  In addition, there is a bunch of old `UserLogin` stuff which also relied on `wgAuth` and `wgExternalAuthType` to inject some logic here and there.  We should get rid of that.  It adds unnecessary complexity.